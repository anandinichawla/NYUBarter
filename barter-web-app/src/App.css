/* button {
  margin: 5px 5px 5px 5px;
  /* margin-right: 150px;

} */

@media only screen and (max-width: 1111px) {
    .search-box{
        display: none!important;
    }
}
@media only screen and (min-width: 1111px) {
    .search-box{
        display: block;
    }
}

@media only screen and (min-width: 580px) {
  .card-img-top{
    width: 30vw;
    height: 25vw;
  }
}

@import url('https://fonts.googleapis.com/css?family=Roboto');
/* @import PreviewPicture from './PreviewPicture'; */

body{
  font-family: "Roboto", sans-serif;
}
.card{
  margin-bottom: 30px;
}
#profileDiv{
  padding-top: 0;
  margin-top: 7px;
}


.btn-primary, .btn-primary:hover, .btn-primary:active, .btn-primary:visited {

  background-color: rgb(255, 63, 85);
  border-color:  rgb(255, 63, 85);
  border-radius: 15px;
  outline: none !important;
}

.form-inline .form-control {
  /* display: inline-block; */
  width: 500px;
  /* vertical-align: middle; */
}
#exampleFormControlSelect1{
  width: 118px;
  margin-right: 5px;
}
#profileImg{
  width: 40px;
  margin-right: 5px;
  border-radius: 50%;
}
#exampleFormControlSelect2{
  width: 115px;
  margin-right: 5px;
}
.homeLink{
  color:rgb(255, 63, 85);
  font-size: 17px;
  /* margin-bottom:150px; */
}

#addItemLabel{
  font-size: 20px;
  background-color:rgb(255, 63, 85);
}

#addItemBtn{
  font-size:15px;
  background-color: white;
  color:  rgb(255, 63, 85);
  border-radius: 2px;
  outline:none !important;
}


.modal-header, .modal-footer{
  background-color: rgb(255, 63, 85);
  color:white;
  font-size: 20px;
  font-family: sans-serif;

}

.col-3{
  margin-top: 20px;
  margin-bottom: 10px;
}


.logoutBtn, .logoutBtn:hover  {
  background-color: white;
  color: rgb(255, 63, 85);
}

.pageHeader{
  /* background-color:  rgb(255, 63, 85); */
  color:rgb(255, 63, 85);;
  font-weight: 20;
  /* height: 90px; */
  /* line-height: 90px; */
  /* text-align: center; */
  height: 80px;
  position: relative;
  left: 10%;
}
.pageText{
  padding-top:15px;
}

.heart{
  border:none;
  color: red;
  float: right;
  font-size: 20px;

}



.barterNav {
  background-color: white;
  color: #ed094d;

}

.barterNav a{
  color:#ed094d;
  font-weight: bold;
}



.input-group-text{
  background: #ed094d !important;
  border: 1px solid #ed094d !important;
  color: white;
  margin: 7.5px 0 0 0;
  /* margin-right:0px; */
  border-top-left-radius: 0px !important;
  border-bottom-left-radius: 0px !important;
  border-top-right-radius: 20px !important;
  border-bottom-right-radius: 20px !important;
  cursor: pointer;
  outline:none !important;
}

.search-box{
    display: inline-flex;
    /* width: 500px; */
    /* width: 60%; */
}
#searchText{
  width:25vw;
}

.form-control {
  width: 60%;
  outline:none !important;
  margin-top: 9px;
  margin-left: 20px;
  border: 1px solid #ed094d !important;
  border-top-left-radius: 20px !important;
  border-bottom-left-radius: 20px !important;
  border-top-right-radius: 0 !important;
  border-bottom-right-radius: 0 !important;
  box-shadow: none;
  outline: none !important;

}

.addItem{
  width: 100%;

  margin-top: 9px;
  margin-left: 20px;
  border: 1px solid #ed094d !important;
  border-top-left-radius: 5px !important;
  border-bottom-left-radius: 5px !important;
  border-top-right-radius: 5px !important;
  border-bottom-right-radius: 5px !important;
  box-shadow: none;
  outline: none !important;

}
#logo{
  width:  150px;
}

.sort{
  position:relative;
  top:20%;
  left: 10%;
  outline: none !important;
}



hr {
  margin: 0;
  border: 1px solid #ed094d !important;

}





.nav-item{
  border-left: 2px solid #fff;
  list-style-type:none;
  padding: 20px 5px;
  text-align: center;
  background-color:  #ed094d;
  cursor: pointer;
  margin: 0;
}

.nav-item button{
  font-size: 16px;
  font-weight: bold;
  color:  #fff;
  text-decoration: none;
  outline: none !important;
}

.dropdown {
  position: relative;
  display: inline-block;
  margin:15px;
}

.dropdown-content {
  display: none;
  position: absolute;
  right: 0;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 1;
  color:darkgrey;
  text-align:center;
}



.dropdown-content a:hover {background-color: #f1f1f1;}

.dropdown:hover .dropdown-content {
  display: block;
}


#logoLogin{
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 50vw;
}
#appstoreLogo{
  margin-left: 9px;
}

#appstoreLogo{
  width:15%;
  position: relative;
  left: 42%;
  top: 20%;

}
