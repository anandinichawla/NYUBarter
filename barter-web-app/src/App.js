import React, { Component } from 'react';
import firebase from 'firebase';
import Rebase from 're-base';
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import Home from './components/Home';
import Interests from './components/Interests';
import Inventory from './components/Inventory';
import { BrowserRouter, Route } from "react-router-dom";

class App extends Component{

  render(){
    return (
      <BrowserRouter>
        <div>
          <Route path="/" component={Home} exact/>
          <Route path="/interests" component={Interests}/>
          <Route path="/inventory" component={Inventory}/>
        </div>
      </BrowserRouter>
    )
  }
}


export default App;
