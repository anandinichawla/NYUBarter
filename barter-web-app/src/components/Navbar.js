import React from 'react';
import { NavLink } from "react-router-dom";
import Link from './Link';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

class Navbar extends React.Component {

  constructor(props) {
    super(props);
    this._onMouseEnter = this._onMouseEnter.bind(this);
    this._onMouseLeave = this._onMouseLeave.bind(this);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter() {
    this.setState({class: STATUS.HOVERED});
  }

  _onMouseLeave() {
    this.setState({class: STATUS.NORMAL});
  }

  render() {
    return (
      <nav>
      <div>
      <ul>
        <li>
          <Link page="/"
                onMouseEnter ={this._onMouseEnter}
                onMouseLeave = {this._onMouseLeave}>
                Home
          </Link>
        </li>


        <li><a href="/interests">Interests</a></li>
        <li><a href="/inventory">Inventory</a></li>
      </ul>
      </div>
      </nav>

    );
  }

}


export default Navbar;
