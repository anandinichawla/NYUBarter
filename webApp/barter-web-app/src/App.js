import React, { Component } from 'react';
import './App.css';
import firebase from './firebase.js';

class App extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      descr: '',
      photoUrl: '',
      username: '',
      dateTime: ''
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  render() {
    return (
      <div className='app'>
        <header>
            <div className='wrapper'>
              <h1>NYU Barter</h1>
              <input type="text" name="search" placeholder="Search for items" />
                <button type="myItems">My Items</button>
                <button type="interestedItems">Interested Items</button>
                <button class="logout">Logout</button>
            </div>
        </header>
        <div className='container'>
          <section className='add-item'>
              <form>
                <input type="text" name="title" placeholder="What item do you want to trade?" onChange={this.handleChange} value={this.state.item} />
                <input type="text" name="descr" placeholder="Describe your item" onChange={this.handleChange} value={this.state.descr} />
                <input type="text" name="photoUrl" placeholder="Add a picture of your item" onChange={this.handleChange} value={this.state.photoUrl} />
                <button>Add Item to Barter</button>
              </form>
          </section>
          <section className='display-item'>
            <div className='wrapper'>
              <ul>
              
              </ul>
            </div>
          </section>
        </div>
      </div>
    );
  }
}
export default App;