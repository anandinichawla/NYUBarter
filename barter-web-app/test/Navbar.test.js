import React from 'react';
import ReactDOM from 'react-dom';
import App from "../src/App";
import Navbar from "../src/components/Navbar";
import renderer from 'react-test-renderer';

// const testFirebase = require('firebase-functions-test')({
//   databaseURL: 'https://https://barterapp-ef89b.firebaseio.com',
//   storageBucket: 'BarterApp.appspot.com',
//   projectId: 'BarterApp',
// }, 'path/to/serviceAccountKey.json');

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};


test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Navbar></Navbar>
  );
  let tree = component.toJSON();
  //console.log(tree);
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});