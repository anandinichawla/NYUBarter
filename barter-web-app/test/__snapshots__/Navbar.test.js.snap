// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<nav>
  <div>
    <ul>
      <li>
        <a
          className="normal"
          href="/"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/interests"
        >
          Interests
        </a>
      </li>
      <li>
        <a
          href="/inventory"
        >
          Inventory
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
